web/src/components/MalteHelpForm.tsx

```diff
@@ -0,0 +1,261 @@
+  height: 1px;
+  border: none;
+`
+
+const SubmitButton = styled(Button)`
+  background-color: ${props => props.theme.colors.textSecondaryColor};
+  color: ${props => props.theme.colors.backgroundColor};
+  border-radius: 100px;
+  width: fit-content;
+  align-self: center;
+  padding: 15px;
+  box-shadow:
+    0 2px 4px rgb(0 0 0 / 15%),
+    0 2px 4px rgb(0 0 0 / 15%);
+
+  :hover {
+    opacity: 80%;
+  }
+
+  :active {
+    box-shadow: none;
+  }
+`
+
+const Form = styled.form`
+  display: flex;
+  flex-direction: column;
+`
+
+const SubmitErrorHeading = styled.h5`
+  margin: 0;
+  font-size: ${props => props.theme.fonts.subTitleFontSize};
+`
+
+const ErrorSendingStatus = styled.div`
+  background-color: ${props => props.theme.colors.invalidInput}35;
+  padding: 20px 10px;
+  margin: 10px 0;
+`
+
+type ContactChannel = 'email' | 'telephone' | 'person'
+type ContactGender = 'male' | 'female' | 'any'
+type SendingStatusType = 'idle' | 'sending' | 'failed' | 'successful'
+
+type MalteHelpFormProps = {
+  submit: () => Promise<void>
+  cityCode: string
+  languageCode: string
+}
+
+const MalteHelpForm = ({ submit, languageCode, cityCode }: MalteHelpFormProps): ReactElement => {
+  const { t } = useTranslation('malteHelpForm')
+  const [sendingStatus, setSendingStatus] = useState<SendingStatusType>('idle')
+  const [submitted, setSubmitted] = useState(false)
+  const [contactChannel, setContactChannel] = useState<ContactChannel>('email')
+  const [email, setEmail] = useState('')
+  const [telephone, setTelephone] = useState('')
+  const [name, setName] = useState('')
+  const [roomNumber, setRoomNumber] = useState('')
+  const [contactType, setContactType] = useState<ContactGender>('any')
+  const [comment, setComment] = useState('')
+  const dashboardRoute = cityContentPath({ languageCode, cityCode })
+
+  const submitHandler = useCallback(
+    (event: SyntheticEvent<HTMLFormElement>) => {
+      const form = event.currentTarget
+      if (!form.reportValidity()) {
+        event.preventDefault()
+        event.stopPropagation()
+        const invalidInputs = Array.from(form.querySelectorAll(':invalid')).sort(
+          (a, b) => a.getBoundingClientRect().top - b.getBoundingClientRect().top,
+        )
+        invalidInputs[0]?.scrollIntoView({ block: 'center', behavior: 'smooth' })
+        setSendingStatus('idle')
+      } else {
+        setSendingStatus('sending')
+        const request = async () => {
+          await submit()
+          setSendingStatus('successful')
+        }
+
+        request().catch(err => {
+          reportError(err)
+          event.preventDefault()
+          setSendingStatus('failed')
+        })
+      }
+    },
+    [submit],
+  )
+
+  if (sendingStatus === 'successful') {
+    return (
+      <Container>
+        <div>{t('submitSuccessful')}</div>
+        <Link to={dashboardRoute}>{t('error:goTo.categories')}</Link>
+      </Container>
+    )
+  }
+
+  return (
```

Comment:
This is quite big, maybe we can extract some code to some reusable components here?

Reply:
I think it's alright. As this file only contains the form. ^^ What would you split/remove?

Reply:
I guess most of it is fine/doesn't make sense to extract. However, I think you could create a `RadioGroup` component similar to what I did in native. That way, it would also be reusable for future use.